credential/validation/validation.go

```diff
@@ -0,0 +1,73 @@
+package validation
+
+import (
+	"fmt"
+
+	"github.com/TBD54566975/ssi-sdk/credential"
+	"github.com/TBD54566975/ssi-sdk/util"
+	"github.com/pkg/errors"
+)
+
+type CredentialValidator struct {
+	validators []Validator
+}
+
+type Validator struct {
+	ID           string
+	ValidateFunc Validate
+}
+
+type (
+	// OptionKey uniquely represents an option to be used in a validator
+	OptionKey string
+)
+
+// Option represents a single option that may be required for a validator
+type Option struct {
+	ID     OptionKey
+	Option any
+}
```

Comment:
Using `any` is discouraged unless there is a very good reason to do so. Can this be updated to received typed params?

Reply:
yes I have a separate ticket to refactor this #352